### PR TITLE
ddl: design doc for `admin pause/resume ddl jobs`

### PR DESCRIPTION
**Pull request by dhysum**
### What problem does this PR solve?
Issue Number:  close https://github.com/pingcap/tidb/issues/40041 https://github.com/pingcap/tidb/issues/18015

Problem Summary: Add Design doc.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: docs/design/imgs/ddl_job_state_machine.png
Comment (Benjamin2037) [inline docs/design/imgs/ddl_job_state_machine.png:1]: This picture could not show correctly.

File: docs/design/2023-04-15-ddl-pause-resume.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+3. Pause the job by
+
+    `admin pause ddl jobs $job_id` ---- $job_id is what you get from earlier step
+
+4. Resume the paused job
+
+    Also, you can find the Paused jobs by :
+
+    `admin show ddl jobs;`
```
Comment (Benjamin2037) [inline docs/design/2023-04-15-ddl-pause-resume.md:92]: Default show job number is 10? should add nums parameter?

File: docs/design/2023-04-15-ddl-pause-resume.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+5. Validate the job's state, and turn it to be Queueing, if `admin resume ...`
+by end user
+
+    5.1 After the job's state changed, the background worker should check it and stop the work
+
+    5.2 No other actions should be taken
+
+### State Machine
+
+![state-machine](./imgs/ddl_job_state_machine.png)
```
Comment (Benjamin2037) [inline docs/design/2023-04-15-ddl-pause-resume.md:70]:
```suggestion
![state-machine](/imgs/ddl_job_state_machine.png)
```

File: docs/design/2023-04-15-ddl-pause-resume.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    4.2 Turn the state to be Pausing
+
+    4.3 Background worker will check the state, and turn the state to be Paused,
+then just return and stay in current stage; And this is what is different from
+`admin cancel ...`
+
+    4.4 Background worker will continue fetching the job and check the state, and
+keep doing nothing until the state changed
+
+    4.5 Specially, the Reorg state could take a long time, we also need to check
```
Comment (Benjamin2037) [inline docs/design/2023-04-15-ddl-pause-resume.md:58]: Should we give the level of waiting time, like seconds? `a long time` is not a good expression.

File: docs/design/2023-04-15-ddl-pause-resume.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+the state of the job, and stop the reorg accordingly
+
+5. Validate the job's state, and turn it to be Queueing, if `admin resume ...`
+by end user
+
+    5.1 After the job's state changed, the background worker should check it and stop the work
```
Comment (Benjamin2037) [inline docs/design/2023-04-15-ddl-pause-resume.md:64]:
```suggestion
    5.1 After the job's state changes, the background worker should check it and resume the job execution.
```

**Review by wjhuang2016 — APPROVED** [review]

File: docs/design/2023-04-15-ddl-pause-resume.md (hunk 1 above)
Comment (dhysum) [reply docs/design/2023-04-15-ddl-pause-resume.md:92]: How about a link to doc on admin show?

File: docs/design/2023-04-15-ddl-pause-resume.md (hunk 4 above)
Comment (dhysum) [reply docs/design/2023-04-15-ddl-pause-resume.md:64]: done

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**